app/controllers/assignments_controller.rb

```diff
@@ -25,10 +25,13 @@ def destroy
   def create
     check_permissions(assignment_params[:user_id].to_i)
     set_wiki { return }
-
+    check_wiki_edu_discouraged_article
```

Comment:
I think this ought to be done a little differently; rather than separately and explicitly calling for this check within the controller, the check against the discouraged titles could be done as part of `AssignmentManager#create_assignment`. This would make it parallel to how we use `AssignmentManager::DuplicateAssignmentError`.

In addition to being simpler, this would also ensure that we do the same `set_clean_title` flow to get the title into the right form before checking it.